invenio_users_resources/notifications/generators.py

```diff
@@ -8,11 +8,22 @@
 
 """User specific generators for notifications."""
 
+import re
 
 from invenio_notifications.models import Recipient
-from invenio_notifications.services.generators import RecipientGenerator
+from invenio_notifications.services.generators import (
+    ConditionalRecipientGenerator,
+    RecipientGenerator,
+)
 from invenio_records.dictutils import dict_lookup
 
+# NOTE: this is the python regex from https://emailregex.com/
+EMAIL_REGEX = re.compile(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)")
+
+
+def _is_email(entity):
```

Comment:
ping @slint 
why wouldn't we check the key instead ?
the receivers were modeled as `{"user": "123"}` or `{"email": "abc@demo.org"}`, we don't have access to this dict here ?
REGEX seems a bit complex :)

Reply:
If it is resolved, the request receiver will only be the e-mail `e.g. {receiver: "abc@demo.org"}`, so we can not perform the simple e-mail check.

Since we also check the unresolved entity for the `IfUserRecipient`, I think it would be ok to do a similar check here as well. The only think we have to keep in mind is to store an unresolved entity somewhere in the context and not resolve it, as would be done here: https://github.com/rekt-hard/invenio-rdm-records/blob/8fc48d39a5786b1ff585a8ea37bb8aad4ae6982b/invenio_rdm_records/notifications/builders.py#L114 (should also be refactored to use the reference resolver registry to create the reference)